### PR TITLE
Kidus/filter articles

### PR DESCRIPTION
**Pull request by kidzegeye**
## Overview

Add article filtering capability based on keywords defined in [volume microservice](https://github.com/cuappdev/volume-microservice).

## Changes Made
Merging this into release as an emergency patch for IOS to get an update out to the App Store.

Added a new field `isFiltered` and filtered articles on all article queries that had `isFiltered: true`. Migration script for existing articles is included. Also added a constant  `IS_FILTER_ACTIVE` to be able to toggle filtering on and off from the backend.

## Test Coverage

Tested through manual testing by checking whether articles with `isFiltered=true` would appear when queried.

## Related PRs or Issues

https://github.com/cuappdev/volume-microservice/pull/12

### CODE REVIEW

File: migrations/add-filter.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module.exports = {
+  async up(db) {
+    let Filter = require("../node_modules/bad-words");
+    const filter = new Filter({emptyList: true});
+    filter.addWords("covid-19","coronavirus", "pandemic","masks","mask")
+    const articles = await db.collection('articles').find({}).toArray();
```
Comment (alanna-zhou) [inline migrations/add-filter.js:6]: can we be consistent about double vs single quotes? is there a linter set in place for this repo?

File: migrations/add-filter.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    articles.map( async (article) => {
+        await db.collection('articles').updateOne({_id: article._id}, {$set: {isFiltered: filter.isProfane(article.title)}});
+        return article;
+    });
+  },
+
+  async down(db) {
```
Comment (alanna-zhou) [inline migrations/add-filter.js:13]: what does this function do? a little confused.. maybe more descriptive name and also a docstring ?

File: src/common/constants.ts [hunk 3]
```diff
@@ -3,3 +3,4 @@ export const IMAGE_ADDRESS = 'https://raw.githubusercontent.com/cuappdev/assets/
 export const MAX_NUM_DAYS_OF_TRENDING_ARTICLES = 30;
 export const IOS = 'IOS';
```
Comment (alanna-zhou) [inline src/common/constants.ts:4]: alphabetize these constants

File: src/repos/ArticleRepo.ts [hunk 4]
```diff
@@ -1,28 +1,44 @@
...
+import { DEFAULT_LIMIT, MAX_NUM_DAYS_OF_TRENDING_ARTICLES, IS_FILTER_ACTIVE } from '../common/constants';
 import { PublicationModel } from '../entities/Publication';
 
+function isArticleFiltered(article: Article){
+  return IS_FILTER_ACTIVE && article.isFiltered;
+}
+
 const getArticleByID = async (id: string): Promise<Article> => {
-  return ArticleModel.findById(new ObjectId(id));
+  return ArticleModel.findById(new ObjectId(id)).then(
```
Comment (alanna-zhou) [inline src/repos/ArticleRepo.ts:12]: you use get getArticleByID tho.. can we keep Id and ID consistent? choose one and stick to it

**Review by alanna-zhou — APPROVED** [review]
only some small issues to be fixed. i'll approve for now since they are all minor changes! great job -- i am outside tho and on my phone so i can't review well (i can only look at cosmetic issues), but overall seems okay :))

File: migrations/add-filter.js (hunk 2 above)
Comment (kidzegeye) [reply migrations/add-filter.js:13]: This is an autogenerated file from migrate-mongo that reverts the changes to the mongo database. I probably can't change its name but I will add a docstring.

File: src/repos/ArticleRepo.ts (hunk 4 above)
Comment (kidzegeye) [reply src/repos/ArticleRepo.ts:12]: findById is a mongoose function

**Review by chalo2000 — APPROVED** [review]
LGTM

**Pull request merged**